lile.go

```diff
@@ -68,47 +110,174 @@ func AddStreamInterceptor(sint grpc.StreamServerInterceptor) {
 	service.StreamInts = append(service.StreamInts, sint)
 }
 
-func Server(r registerImplementation) {
-	service.GRPCImplementation = r
+func Run() {
+	// add to registry
+	if service.Config.UsesRegistry() {
+		rc, rcErr := GetRegistryClient()
+		if rcErr == nil {
+			err := rc.Register(service.ID, service.Name, service.Config.Service.Port, &api.AgentServiceCheck{
+				CheckID:                        "",
+				Name:                           "",
+				Args:                           nil,
+				Script:                         "",
+				DockerContainerID:              "",
+				Shell:                          "",
+				Interval:                       "",
+				Timeout:                        "",
+				TTL:                            "",
+				HTTP:                           "",
+				Header:                         nil,
+				Method:                         "",
+				TCP:                            "",
+				Status:                         "",
+				Notes:                          "",
+				TLSSkipVerify:                  false,
+				GRPC:                           "",
+				GRPCUseTLS:                     false,
+				DeregisterCriticalServiceAfter: "",
+			})
+
+			if err != nil {
+				logrus.Errorf("Failed to register service at '%s'. error: %v", service.Config.RegistryAddress, err)
+			} else {
+				logrus.Infof("Regsitered service '%s' at consul.", service.ID)
```

Comment:
Do the logging in the consul implementation I think!

Reply:
Sure, why not!